src/Nethermind/Nethermind.Init/Steps/Migrations/ReceiptFixMigration.cs

```diff
@@ -47,14 +47,15 @@ public void Run()
                 _cancellationTokenSource = new CancellationTokenSource();
                 CancellationToken cancellationToken = _cancellationTokenSource.Token;
 
-                MissingReceiptsFixVisitor visitor = new MissingReceiptsFixVisitor(
-                    syncConfig.PivotNumberParsed,
+                MissingReceiptsFixVisitor visitor = new(
+                    syncConfig.AncientReceiptsBarrierCalc,
```

Comment:
`syncConfig.PivotNumberParsed` delegates to `syncConfig.PivotNumber`, which is **not a read-only field**. The pivot number is modified here:

https://github.com/NethermindEth/nethermind/blob/52d7addb56dc2d9d91dd0f8b5844c8d3fae2c521/src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PivotUpdator.cs#L77

then, by the time that this code gets executed, the pivot number is not what the user originally configured.

---

Either way, we should use the `AncientReceiptsBarrier` value for this operation.